imod/prepare/cleanup.py

```diff
@@ -209,11 +211,80 @@ def cleanup_ghb(
     return _cleanup_robin_boundary(idomain, output_dict)
 
 
-def cleanup_wel(wel_ds: GridDataset):
+def cleanup_wel(
+    wells: pd.DataFrame,
+    top: GridDataArray,
+    bottom: GridDataArray,
+    minimum_thickness=0.05,
+):
     """
-    Clean up wells
+    Clean up dataframe with wells, fixes some common mistakes in the following
+    order:
+
+    - Wells outside grid bounds are dropped
+    - Filters above surface level are set to surface level
+    - Drop wells with filters entirely below base
+    - Clip filter screen_bottom to model base
+    - Clip filter screen_bottom to screen_top
+    - Well filters thinner than minimum thickness are made point filters
 
-    - Removes wells where the screen bottom elevation exceeds screen top.
+    Parameters
+    ----------
+    wells: pandas.Dataframe
+        Dataframe with wells to be cleaned up. Requires columns ``"x", "y",
+        "id", "screen_top", "screen_bottom"``
+    top: xarray.DataArray | xugrid.UgridDataArray
+        Grid with model top
+    bottom: xarray.DataArray | xugrid.UgridDataArray
+        Grid with model bottoms
+    minimum_thickness: float
+        Minimum thickness, filter thinner than this thickness are set to point
+        filters
+
+    Returns
+    -------
+    pandas.DataFrame
+        Cleaned well dataframe.
     """
-    deactivate = wel_ds["screen_top"] < wel_ds["screen_bottom"]
-    return wel_ds.where(~deactivate, drop=True)
+    validate_well_columnnames(
+        wells, names={"x", "y", "id", "screen_top", "screen_bottom"}
+    )
+
+    # 1. Locate wells, wells outside grid bounds are dropped
```

Comment:
Instead of using comments to structure your code you could also extract each part to a separate method

```
def cleanup_wells(...):
  ...
  cleaned_wells = _locate_wells(...)
  cleaned_wells = _clip_top_of_wells(...)
  cleaned_wells = _drop_wells_below_base(...)
  cleaned_wells = _convert_inverted_wells_to_point_wells(...)
  cleaned_wells = _convert_zero_thickness_wells_to_point_wells(...)

  return cleaned_wells
```

Reply:
Good suggestion, done